llvm/lib/IR/IRBuilder.cpp

```diff
@@ -918,10 +918,11 @@ CallInst *IRBuilderBase::CreateUnaryIntrinsic(Intrinsic::ID ID, Value *V,
   return createCallHelper(Fn, {V}, Name, FMFSource);
 }
 
-CallInst *IRBuilderBase::CreateBinaryIntrinsic(Intrinsic::ID ID, Value *LHS,
-                                               Value *RHS,
-                                               Instruction *FMFSource,
-                                               const Twine &Name) {
+Value *IRBuilderBase::CreateBinaryIntrinsic(Intrinsic::ID ID, Value *LHS,
+                                            Value *RHS, Instruction *FMFSource,
+                                            const Twine &Name) {
+  if (auto *V = Folder.FoldBinaryIntrinsic(ID, LHS, RHS))
```

Comment:
Can we pass the argument `FMFSource` into `Folder.FoldBinaryIntrinsic`?

Reply:
It is not immediately clear to me how it can be used, given that `FoldBinaryIntrinsic` returns a `Constant` and from my understanding the `FMFSource` is used to copy FMF flags when a new `CallInst` is created.

Reply:
We can use FMF flags to further fold expressions in `InstSimplifyFolder`.
E.g., `minimum(X, -inf) -> -inf when FMF nnan is set`.